test/semantics/label01.F90

```diff
@@ -0,0 +1,226 @@
+! you may not use this file except in compliance with the License.
+! You may obtain a copy of the License at
+!
+!     http://www.apache.org/licenses/LICENSE-2.0
+!
+! Unless required by applicable law or agreed to in writing, software
+! distributed under the License is distributed on an "AS IS" BASIS,
+! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+! See the License for the specific language governing permissions and
+! limitations under the License.
+
+! RUN: f18 < %s | FileCheck %s
+! CHECK-NOT: 
+
+! these are the conformance tests
+! define STRICT_F18 to eliminate tests of features not in F18
+! define ARCHAIC_FORTRAN to add test of feature found in Fortran before F95
+
+subroutine sub00(a,b,n,m)
+  real a(n)
+  real :: b(m)
+1 print *, n, m
+1234 print *, a(n), b(1)
+99999 print *, a(1), b(m)
+end subroutine sub00
+
+subroutine do_loop01(a,n)
+  real, dimension(n) :: a
+  do 10 i = 1, n
+     print *, i, a(i)
+10   continue
+end subroutine do_loop01
+
+subroutine do_loop02(a,n)
+  real, dimension(n,n) :: a
+  do 10 j = 1, n
+     do 10 i = 1, n
+        print *, i, j, a(i, j)
+10      continue
+end subroutine do_loop02
+
+#ifndef STRICT_F18
+subroutine do_loop03(a,n)
+  real, dimension(n) :: a
+  do 10 i = 1, n
+10   print *, i, a(i)		! extension (not f18)
+end subroutine do_loop03
+
+subroutine do_loop04(a,n)
+  real :: a(n,n)
+  do 10 j = 1, n
+     do 10 i = 1, n
+10      print *, i, j, a(i, j)	! extension (not f18)
+end subroutine do_loop04
+
+subroutine do_loop05(a,n)
+  real a(n,n,n)
+  do 10 k = 1, n
+     do 10 j = 1, n
+        do 10 i = 1, n
+10         print *, a(i, j, k)	! extension (not f18)
+end subroutine do_loop05
+#endif
+
+subroutine do_loop06(a,n)
+  real, dimension(n) :: a
+  loopname: do i = 1, n
+     print *, i, a(i)
+     if (i .gt. 50) then
+678     exit
+     end if
+  end do loopname
+end subroutine do_loop06
+
+subroutine do_loop07(a,n)
+  real, dimension(n,n) :: a
+  loopone: do j = 1, n
+     looptwo: do i = 1, n
+        print *, i, j, a(i, j)
+     end do looptwo
+  end do loopone
+end subroutine do_loop07
+
+subroutine do_loop08(a,b,n,m,nn)
+  real, dimension(n,n) :: a
+  real b(m,nn)
+  loopone: do j = 1, n
+     condone: if (m .lt. n) then
+        looptwo: do i = 1, m
+           condtwo: if (n .lt. nn) then
+              b(m-i,j) = s(m-i,j)
+              if (i .eq. j) then
+                 goto 111
+              end if
+           else
+              cycle loopone
+           end if condtwo
+        end do looptwo
+     else if (n .lt. m) then
+        loopthree: do i = 1, n
+           condthree: if (n .lt. nn) then
+              a(i,j) = b(i,j)
+              if (i .eq. j) then
+                 return
+              end if
+           else
+              exit loopthree
+           end if condthree
+        end do loopthree
+     end if condone
+  end do loopone
+111 print *, "done"
+end subroutine do_loop08
+
+#ifndef STRICT_F18
```

Comment:
Should the f18 preprocessor automatically define this macro?  Probably not, but something to consider.

Reply:
Good idea. I think I'll defer for now, but I'll make a note.